packages/gatsby-plugin-netlify/src/gatsby-node.js

```diff
@@ -34,7 +34,7 @@ exports.onPostBuild = async ({ store, pathPrefix }, userPluginOptions) => {
   let rewrites = []
   if (pluginOptions.generateMatchPathRewrites) {
     const { pages } = store.getState()
-    rewrites = pages
+    rewrites = Array.from(pages.values())
```

Comment:
You should be able to gain some more speed using reduce:

```js
rewrites = Array.from(pages.values())
	.reduce((acc, page) => {
  	if (page.matchPath && page.matchPath !== page.path) {
    	acc.push({
        fromPath: page.matchPath,
        toPath: page.path,
      })
    }
    
    return acc;
  }, [])
```

Reply:
Why would reducing be faster than filtering?

Reply:
You get to avoid using map to iterate through the generated array from the filter. With reduce you generate the end result in a single iteration.

Reply:
Hmm yeah. Generally there'll only a handful of pages w/ `matchPath` so making the change should make a negligible difference 🤷‍♂️ 

That being said, if you'd like to make the PR, happy to take it! Thanks for reading through the PR!